pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/ComputeStep.java

```diff
@@ -293,8 +293,8 @@ private Long getAvroTimestampMillis(Object value, LogicalType logicalType) {
   private Integer getAvroDate(Object value, LogicalType logicalType) {
     validateLogicalType(value, logicalType, Date.class, LocalDate.class);
     return (value instanceof LocalDate)
-        ? (int) ((LocalDate) value).toEpochDay()
-        : (int) (((Date) value).getTime() / MILLIS_PER_DAY);
+        ? Math.toIntExact(((LocalDate) value).toEpochDay())
```

Comment:
Max year that can be represented by epoch days as integer is "5881580-07-11"! After that, our transformation we overflow :D

Reply:
Yes... I prefer to fail here than outputing a random value.

Reply:
I'd be concerned about the world if this code is still in use by the year 5881580